independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/AsmUtilCopy.java

```diff
@@ -224,21 +211,6 @@ private static void toSignature(StringBuilder sb, List<TypeVariable> typeParamet
             sb.append(">");
     }
 
-    /**
-     * Returns true if the given class has type parameters or if its superclass or superinterfaces require a signature
-     */
-    public static boolean needsSignature(ClassInfo klass) {
```

Comment:
Do we really need to remove this method? It looks orthogonal to the issue at hand?

But I'll let @mkouba decide about all this anyway.

Reply:
It's not orthogonal; it was used in `ClientProxyGenerator` to know whether we need to create a signature for a proxy. We used to pass the proxied class as a parameter. But that was wrong: a proxy class needing a signature is not the same as its parent class needing a signature.

For example:

```java
class MyClass implements Supplier<Integer> {
}

class MyClass_Proxy extends MyClass {
}
```

The proxied class (`MyClass`) needs a signature, since it extends a parameterized type, but the proxy class (`MyClass_Proxy`) doesn't.

Since this method was being misused, and the only use was in `ClientProxyGenerator`, and it very much looks like internal code, I think it should be removed.